cogs/helpers.py

```diff
@@ -563,6 +564,94 @@ async def colour(self, ctx, *, arg: str):
         fn = "{}.{}".format(match.group(1), ext)
         await ctx.send(file=discord.File(fp=buffer, filename=fn))
 
+    @commands.Cog.listener()
+    async def on_member_remove(self, member):
+        roles_id = [
+            role.id for role in member.roles if role.name != "@everyone"
+        ]
+        if roles_id:
+            conn = sqlite3.connect(self.bot.config.db_path)
+            c = conn.cursor()
+            # store roles as a string of IDs separated by spaces
+            t = (member.id, ' '.join(str(e) for e in roles_id))
+            c.execute('REPLACE INTO PreviousRoles VALUES (?, ?)', t)
+            conn.commit()
+            conn.close()
+
+    @commands.command(aliases=['previousroles', 'giverolesback', 'rolesback'])
+    async def previous_roles(self, ctx, user: discord.Member):
+        """Show the list of roles that a user had before leaving, if possible.
+        A moderator can click the OK react on the message to give these roles back
+        """
+        conn = sqlite3.connect(self.bot.config.db_path)
+        c = conn.cursor()
+        fetched_roles = c.execute(
+            'SELECT Roles FROM PreviousRoles WHERE ID = ?',
+            (user.id, )).fetchone()
+        # the above returns a tuple with a string of IDs separated by spaces
+        if fetched_roles is not None:
+            roles_id = list(fetched_roles[0].split(" "))
+            valid_roles = []
+            for role in roles_id:
+                try:
+                    valid_roles.append(
+                        self.bot.get_guild(self.bot.config.server_id).get_role(
+                            int(role)))
+                except AttributeError:
+                    pass
+
+            roles_name = [
+                "[{}] {}\n".format(i + 1, role.name)
+                for i, role in enumerate(valid_roles)
```

Comment:
should have been better reviewed the first time, my bad:
```python
for i, role in enumerate(valid_roles, 1)
```
to avoid adding 1 each time

Reply:
Resolved in e375496